test/scenarios.py

```diff
@@ -4057,6 +4057,187 @@
             ),
         },
     ),
+    Scenario(
+        "scenario21-3",
+        "Cancel the directory transfer in flight, then resume transfer. Expect it to be resumed properly",
+        {
+            "ren": ActionList(
+                [
+                    action.ConfigureNetwork(),
+                    action.WaitForAnotherPeer(),
+                    action.NewTransfer("172.20.0.15", ["/tmp/nested/big"]),
+                    action.Wait(
+                        event.Queued(
+                            0,
+                            {
+                                event.File(
+                                    FILES["nested/big/testfile-01"].id,
+                                    "big/testfile-01",
+                                    10485760,
+                                ),
+                                event.File(
+                                    FILES["nested/big/testfile-02"].id,
+                                    "big/testfile-02",
+                                    10485760,
+                                ),
+                            },
+                        )
+                    ),
+                    action.WaitRacy(
+                        [
+                            event.Start(0, FILES["nested/big/testfile-01"].id),
+                            event.Start(0, FILES["nested/big/testfile-02"].id),
+                        ]
+                    ),
+                    action.Wait(event.FinishTransferCanceled(0, True)),
+                    # new transfer
+                    action.NewTransfer("172.20.0.15", ["/tmp/nested/big"]),
+                    action.Wait(
+                        event.Queued(
+                            1,
+                            {
+                                event.File(
+                                    FILES["nested/big/testfile-01"].id,
+                                    "big/testfile-01",
+                                    10485760,
+                                ),
+                                event.File(
+                                    FILES["nested/big/testfile-02"].id,
+                                    "big/testfile-02",
+                                    10485760,
+                                ),
+                            },
+                        )
+                    ),
+                    action.WaitRacy(
+                        [
```

Comment:
But this actually sends two separate requests and tests that those two requests were sent properly. We don't even test that there should be progress which is equal/higher to the one that was previously reported. The IDs also will differ so the marker file will be different so it shouldn't really resume?